### PR TITLE
Download single file from server in a single click

### PR DESCRIPTION
**Pull request by vivonk**
## Feature Implementation

After this commit changes, a user will able to download the files from the server to the local system.

## Description
Implemented download feature on backend and provided single click download feature on the front end

## Related Issue

#453 

## Motivation and Context

Enhancing the Bassa and implement the main feature which makes Bassa better
## How Has This Been Tested?

Tested with Safari, Mozilla, and Chrome. Tested on cross-platforms.
Run standard tests: `npm test`

## Screenshots (In case of UI changes):
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## TODO:
- [ ] Add multiple file download feature
- [ ] Write Unit tests for changes
- [ ] Add Auth System to the file sending API

### CODE REVIEW

File: components/core/file_sender/sender.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        download_name = get_download_name_from_id(id)
+    except Exception as e:
+        logging.error(" get file API (/api/get/file) got wrong arguments, thrown an error >> " + str(e) + " <<")
+        return Response("error", status=200)
+
+    try:
+        return send_file(filename_or_fp=download_path, attachment_filename=download_name,
+                         as_attachment=True)
+    except Exception as e:
+        logging.error("File sending has a exception. We got >> "+str(e)+" << error on file sending")
```
Comment (agentmilindu) [inline components/core/file_sender/sender.py]: Let's not get the `str(e)` to a middle of a sentence, but let's have it as the end f the sentence.

Comment (vivonk) [reply components/core/file_sender/sender.py]: done 👍

File: components/core/file_sender/sender.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        download_name = get_download_name_from_id(id)
+    except Exception as e:
+        logging.error(" get file API (/api/get/file) got wrong arguments, thrown an error :: %s" % e)
+        return Response("error", status=200)
+
+    try:
+        return send_file(filename_or_fp=download_path, attachment_filename=download_name,
+                         as_attachment=True)
+    except Exception as e:
+        logging.error("File sending has a exception. We got :: %s on file sending" % e)
```
Comment (agentmilindu) [inline components/core/file_sender/sender.py]: I think you have missed this error to get to the end.

Comment (vivonk) [reply components/core/file_sender/sender.py]: I think it's correct? Are you saying that this error should be placed after last return?
Actually error returns some text in response so it's that return statement

Comment (agentmilindu) [reply components/core/file_sender/sender.py]: No, this is relevant to my previous comment about placing the error object in the log message. 

```
logging.error("File sending has a exception. When sending file, we got :: %s" % e)
```

Comment (vivonk) [reply components/core/file_sender/sender.py]: oh, sorry I got it wrong. So sorry, ok let me change that

**Pull request merged**